src/backend/stream.rs

```diff
@@ -414,6 +414,47 @@ impl<'ctx> PulseStream<'ctx> {
             }
         }
 
+        fn get_default_sink_name(
+            mainloop: &pulse::ThreadedMainloop,
+            context: &pulse::Context,
+        ) -> String {
+            struct CallbackData<'a> {
+                mainloop: &'a pulse::ThreadedMainloop,
+                default_sink_name: String,
+            }
+
+            fn server_info_cb(
+                _context: &pulse::Context,
+                info: Option<&pulse::ServerInfo>,
+                u: *mut c_void,
+            ) {
+                let r = unsafe { ptr::NonNull::new(u as *mut CallbackData).unwrap().as_mut() };
+
+                let name = unsafe { CStr::from_ptr(info.unwrap().default_sink_name) };
```

Comment:
Should be ALWAYS valid, because we control when the callback is called, not the Pulse API. By that I mean me make sure that the PulseAudio API calls our callback ONLY WHEN we want it, otherwise it'd be a massive UB hole to patch.

Reply:
Not that familiar with Rust as you can notice, but I assume that the callback function would be invalid if we leave the scope (by that I mean the ``get_default_sink_name`` function)?